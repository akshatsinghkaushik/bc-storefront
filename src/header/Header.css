.header {
  padding: 0.5rem 1rem 0.5rem 1rem;
  height: 3rem;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid rgb(223, 222, 222);
  border-width: 100%;
}

.header-links {
  transform: translateX(-10%);
}

.header-link {
  margin: 0 1rem 0 1rem;
}

.my-cart {
  display: inline;
  letter-spacing: 1px;
  font-size: 13px;
}
