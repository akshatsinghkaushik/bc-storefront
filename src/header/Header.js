import React, { useState, useContext, useEffect, useRef } from 'react';
import './Header.css';
import CartPopUp from './CartPopUp';
import CartContext from '../context/CartContext';

const Header = ({}) => {
  const [cartPopUp, setCartPopUp] = useState(false);
  const cartContext = useContext(CartContext);

  const node = useRef();
  const link = useRef();

  const handleClick = (e) => {
    if (node.current && link.current) {
      if (node.current.contains(e.target) || link.current.contains(e.target)) {
        // inside click
        return;
      }
      // outside click
      setCartPopUp(false);
    }
  };

  useEffect(() => {
    // add when mounted
    document.addEventListener('click', handleClick);
    // return function to be called when unmounted
    return () => {
      document.removeEventListener('click', handleClick);
    };
  }, []);

  return (
    <>
      <header className="header">
        <div className="header-logo">
          <img src="/media/logo.png" alt="Site Logo" />
        </div>
        <div
          className="header-links"
          style={{ transform: 'translateX(0.5rem)' }}
        >
          <a
            className="header-link"
            href="/"
            style={{ textDecoration: 'none', color: 'black' }}
          >
            HOME
          </a>
          <a
            className="header-link"
            href="/"
            style={{ textDecoration: 'none', color: 'black' }}
          >
            SHOP
            <img
              style={{
                height: '0.7rem',
                width: '0.7rem',
                marginLeft: '0.3rem',
              }}
              src={`/media/chevron-down-solid.svg`}
            />
          </a>
          <a
            className="header-link"
            href="/"
            style={{ textDecoration: 'none', color: 'black' }}
          >
            JOURNAL
          </a>
          <a
            className="header-link"
            href="/"
            style={{ textDecoration: 'none', color: 'black' }}
          >
            MORE
            <img
              style={{
                height: '0.7rem',
                width: '0.7rem',
                marginLeft: '0.3rem',
              }}
              src={`/media/chevron-down-solid.svg`}
            />
          </a>
        </div>
        <div className="cart-icon">
          <div
            onClick={() => setCartPopUp(!cartPopUp)}
            style={{ cursor: 'pointer' }}
            ref={link}
          >
            <p
              style={
                cartPopUp
                  ? {
                      borderBottom: '1.3px solid black',
                      paddingBottom: '0.3rem',
                    }
                  : {}
              }
              className="my-cart"
            >
              MY CART
              {cartContext.cartMapArr.length === 0
                ? ''
                : ` (${cartContext.getCartQuantity()})`}
            </p>

            <img
              style={{
                height: '0.7rem',
                width: '0.7rem',
                marginLeft: '0.5rem',
              }}
              src={
                cartPopUp
                  ? `/media/chevron-right-solid.svg`
                  : `/media/chevron-down-solid.svg`
              }
            />
          </div>
          <CartPopUp
            ref={node}
            cartPopUp={cartPopUp}
            setCartPopUp={setCartPopUp}
          />
        </div>
      </header>
    </>
  );
};

export default Header;
