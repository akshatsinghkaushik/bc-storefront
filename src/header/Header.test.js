import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';
import MockCartProvider from '../tests/MockCartProvider';

it('Header Renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MockCartProvider>
          <Header />
        </MockCartProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
