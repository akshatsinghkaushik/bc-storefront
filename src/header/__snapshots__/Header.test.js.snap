// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="header-logo"
  >
    <img
      alt="Site Logo"
      src="/media/logo.png"
    />
  </div>
  <div
    className="header-links"
    style={
      Object {
        "transform": "translateX(0.5rem)",
      }
    }
  >
    <a
      className="header-link"
      href="/"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
      HOME
    </a>
    <a
      className="header-link"
      href="/"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
      SHOP
      <img
        src="/media/chevron-down-solid.svg"
        style={
          Object {
            "height": "0.7rem",
            "marginLeft": "0.3rem",
            "width": "0.7rem",
          }
        }
      />
    </a>
    <a
      className="header-link"
      href="/"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
      JOURNAL
    </a>
    <a
      className="header-link"
      href="/"
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
      MORE
      <img
        src="/media/chevron-down-solid.svg"
        style={
          Object {
            "height": "0.7rem",
            "marginLeft": "0.3rem",
            "width": "0.7rem",
          }
        }
      />
    </a>
  </div>
  <div
    className="cart-icon"
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <p
        className="my-cart"
        style={Object {}}
      >
        MY CART
        
      </p>
      <img
        src="/media/chevron-down-solid.svg"
        style={
          Object {
            "height": "0.7rem",
            "marginLeft": "0.5rem",
            "width": "0.7rem",
          }
        }
      />
    </div>
    <div
      className="cart-popup"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="popup-items-container"
      >
        <div
          style={
            Object {
              "padding": "1rem",
              "textAlign": "center",
            }
          }
        >
          No Items in Cart
        </div>
      </div>
      <hr
        style={
          Object {
            "height": "1px",
            "margin": "0",
            "width": "100%",
          }
        }
      />
      <div
        className="popup-footer"
        style={
          Object {
            "margin": "1rem 1rem 0 1rem",
          }
        }
      >
        <p>
          TOTAL
        </p>
        <p>
          $
          0.00
        </p>
      </div>
      <div
        className="popup-footer"
      >
        <a
          className="popup-footer-view"
          href="/cart"
          onClick={[Function]}
        >
          VIEW CART
        </a>
        <button
          className="popup-footer-checkout"
        >
          CHECKOUT
        </button>
      </div>
    </div>
  </div>
</header>
`;
